build_tools/benchmarks/collect_compilation_statistics.py

```diff
@@ -15,13 +15,20 @@
 import os
 import re
 import zipfile
-
+import sys
 from dataclasses import asdict
+from pathlib import PurePath, Path
 from typing import BinaryIO, Dict, Optional, TextIO
-from pathlib import PurePath
-
-from common.benchmark_definition import CompilationInfo, CompilationResults, CompilationStatistics, ModuleComponentSizes, get_git_commit_hash
+# Add build_tools python dir to the search path.
+sys.path.append(
```

Comment:
Hmm, this is not nice. Is there another way to avoid having this? What about

* Putting `build_tools` as the common parent module?, or
* Moving Python scripts in `benchmarks` under the `python` directory?

Reply:
Yeah, this is the part I need suggestions for. AFAIK, if you run `python build_tools/benchmarks/xyz.py`, the script won't be able to find the modules above the directory `build_tools/benchmarks`, because they are not in the search path.

We have python scripts outside `benchmarks` that want to share code. So unless we put all the scripts under the `python` directory, it won't solve the problem. And even if we agree to put all scripts under `python` directory, all executable scripts will need to be right under `python`; otherwise they still can't find the common modules in their parent directories.

Some alternative ways I know are:

- Pack the build tools scripts into a python package. But this will require everyone including CI to install the package before running the tools (and we have CMake rules depending on these scripts).
- Require everyone to run scripts with `python -m`. If you run `python -m build_tools.benchmarks.xyz`, the `xyz.py` will be able to use relative paths to import the modules under `build_tools` or `build_tools/python`.
- Require everyone to run scripts with `PYTHONPATH=build_tools/python` in the env.

Reply:
Of all of these, I hate the last one the least, but I actually think that what you have now is not too bad. One way or another we have to put information about the local file system into Python, either via an environment variable or by Python figuring it out itself

Reply:
Yeah. I don't have better suggestions anyway so unblocking this from my side. :)

Reply:
Hi @stellaraccident , do you have some suggestions on this? : )

Reply:
This kind of thing is pretty par for the course for in-place, in-repo utilities.

My guidance:

* If there are just going to be a couple of these (ever), then just do the path munging at the top of each.
* Otherwise, commit to only having each "executable" python script be a trampoline that does two things: a) munge the path, b) import the tool, c) delegate to the tool's main().

If going to the trouble of organizing this, I'd recommend the second option. Your scripts will have some repetition, but they will basically be *only* trampolines, so they should all be roughly the same.